src/Chartjs/assets/src/controller.ts

```diff
@@ -10,7 +10,9 @@
 'use strict';
 
 import { Controller } from '@hotwired/stimulus';
-import Chart from 'chart.js/auto';
+import { Chart, registerables } from 'chart.js';
+
+Chart.register(...registerables);
```

Comment:
```suggestion
// ChartJs 3.x
if (registerables != undefined) {
    Chart.register(...registerables);
}
```

It's kinda a too simple fix, but it makes the test-suite pass (for `3.x` and `4.x`). In the `3.x` version the `registerables` variable is `undefined` where as in `4.x` it consists of an list of 'registerables' :)

```js
[
  [Object: null prototype] {
    BarController: [class BarController extends DatasetController] {
      id: 'bar',
      defaults: [Object],
      overrides: [Object]
    },
    BubbleController: [class BubbleController extends DatasetController] {
      id: 'bubble',
      defaults: [Object],
      overrides: [Object]
    },
    DoughnutController: [class DoughnutController extends DatasetController] {
      id: 'doughnut',
      defaults: [Object],
      descriptors: [Object],
      overrides: [Object]
    }
    ...
```

I currently don't have a project where I use a plugin, so that should be tested/verified. Maybe we should add a simple test to the suite, where we test using a plugin?

An other solution I came across also seems to work, but then I'd prefer the previous solution:
```js
import { Chart } from 'chart.js';
import * as Chartjs from 'chart.js';

const controllers = Object.values(Chartjs).filter(
  (chart) => chart.id !== undefined
);

Chart.register(...controllers);
```

Reply:
That's a good lead thank you. We'll need to check if everything works the same ... (and i still don't understand how it can be undefined as it is exported in the 3.9.x source code https://cdn.jsdelivr.net/npm/chart.js@3.9.0/dist/chart.esm.js 

And i have the impression all the problem is indeed that the esm version is not the one used (maybe only in test)

Reply:
If you look at "chartjs/auto" https://cdn.jsdelivr.net/npm/chart.js@3.9.0/auto/auto.mjs

Reply:
Yeah that is pretty strange, it could specifically be testsuite related. I'll try to test it in a real project.